corehq/ex-submodules/couchforms/models.py

```diff
@@ -368,6 +368,7 @@ def from_xform_instance(cls, instance, error_message, with_new_id=False):
         instance.__class__ = XFormError
         instance.doc_type = 'XFormError'
         instance.problem = error_message
+        instance.orig_id = None
```

Comment:
Why is this needed? Given that you're checking for the attribute at

https://github.com/dimagi/commcare-hq/blob/17130ad420fad472c02172979c0268007bb4f1eb/corehq/form_processor/submission_post.py#L356

Reply:
This was the hack I mentioned. We're taking an `XFormInstance` and relabelling it as an `XFormError` without running the proper initialization. Surprisingly, this means that running `hasattr(form, 'orig_id')` fails.